pkg/server/server.go

```diff
@@ -465,14 +468,17 @@ func (s *ProxyServer) serveRecvBackend(stream agent.AgentService_ConnectServer,
 			} else {
 				err := client.send(pkt)
 				s.PendingDial.Remove(resp.Random)
-				if err != nil {
+				if resp.Error != "" {
```

Comment:
I would move this condition before the client.send(), i.e.,

```golang
if resp.Error != "" {
  klog.Warningf("<<< DIAL_RSP contains error: %v", resp.Error)
}
err := client.send(pkt)
s.PendingDial.Remove(resp.Random)
if err != nil {
  klog.Warningf("<<< DIAL_RSP send to client stream error: %v", err)
} else {
  ...
}
```

Reply:
Took a slightly modified approach of your suggestion.